Network/Socket/Syscall.hs

```diff
@@ -120,6 +121,15 @@ socket family stype protocol = E.bracketOnError create c_close $ \fd -> do
     unsetIPv6Only _ = return ()
 #endif
 
+    setDontFragment s = when (family == AF_INET) $
+#if HAVE_DECL_IP_DONTFRAG || HAVE_DECL_IP_MTU_DISCOVER
+      setSocketOption s DontFragment 1
+#else
```

Comment:
These four lines (`#else` clause) are not necessary

Reply:
if there are OSs/kernels which don't have those sockoption, I guess the compilation fails like below.

```
Network/Socket/Syscall.hs:138:1: error: [GHC-58481]
    parse error (possibly incorrect indentation or mismatched brackets)
    |
138 | bind :: SocketAddress sa => Socket -> sa -> IO ()
    | ^
```

I followed the way ipv6only option does in the `#else` clause.
do you still think it's unnecessary ?

Reply:
Never mind.
I misunderstand that you are using `setDontFragment` here but you are defining it here.